### PR TITLE
balenaOS: warn about a custom device type (custom OS)

### PR DESCRIPTION
**Pull request by xginn8**
Connects-to: #147
Connects-to: #112
Signed-off-by: Matthew McGinn <matthew@balena.io>
Change-type: patch

### CODE REVIEW

File: scripts/checks.sh [hunk 1]
```diff
@@ -62,18 +62,22 @@ function check_balenaOS()
 	# VERSION="1.24.0"
 	# PRETTY_NAME="Resin OS 1.24.0"
 	if grep -q -e '^VERSION="1.*$' -e '^PRETTY_NAME="Resin OS 1.*$' /etc/os-release; then
-		log_status "${BAD}" "${FUNCNAME[0]}" "resinOS 1.x is now completely deprecated"
+		log_status "${BAD}" "${FUNCNAME[0]}" "ResinOS 1.x is now completely deprecated"
 	else
 		# shellcheck disable=SC1091
 		source /etc/os-release
+		if [[ "${DEVICE_TYPE}" != "${SLUG}" ]]; then
```
Comment (ZubairLK) [inline scripts/checks.sh]: Where is `${DEVICE_TYPE}` coming from here

Comment (xginn8) [reply scripts/checks.sh]: In line 5 we source `/usr/sbin/resin-vars` which provides this envvar.

Comment (ZubairLK) [reply scripts/checks.sh]: hmm. I wasn't aware that `SLUG` in `etc/os-release` and `DEVICE_TYPE` in config.json are different..

Comment (xginn8) [reply scripts/checks.sh]: I think that's only true iff the device type is custom (`SLUG` indicating the base type, `DEVICE_TYPE` indicating the custom type). That is what's indicated in the attached issues, at least.

**Comment by xginn8** [comment]
@balena-ci retest

File: scripts/checks.sh (hunk 1 above)
Comment (ZubairLK) [reply scripts/checks.sh]: I see. TIL

**Comment by xginn8** [comment]
@balena-ci retest

**Pull request merged**